client/lobby/CBonusSelection.cpp

```diff
@@ -455,12 +455,22 @@ void CBonusSelection::restartMap()
 
 void CBonusSelection::increaseDifficulty()
 {
+	// Rely on server to saturate maximum difficuly level
+	// instead of wrap around:
+	//     ui8 difficulty = hardest;
+	//     min<ui8>(difficulty + 1, hardest) -> hardest (ok)
 	CSH->setDifficulty(CSH->si->difficulty + 1);
 }
 
 void CBonusSelection::decreaseDifficulty()
 {
-	CSH->setDifficulty(CSH->si->difficulty - 1);
+	// Rely on client to saturate minimum difficuly level
+	// instead of wrap-around:
+	//     ui8 difficulty = 0; // easiest
+	//     max<ui8>(difficulty - 1, 0) -> 255 (overflow)
+	//     max<ui8>(255, hardest) -> hardest (wrap-around)
```

Comment:
I personally think that one comment here is enough. Something like
// avoid negative overflow (0 - 1 = 255)
right before if. I personally believe that if code needs comments - something bad with overall design. So I would change si->dificulty to int. But you solution is easier and less impactful. So I suggest to simplify the comment. It will be enough for code reader to understand why if is needed

Reply:
Sounds good. Left only `// avoid negative overflow (0 - 1 = 255)` comment.